src/main/java/com/singlestore/fivetran/destination/SingleStoreDestinationServiceImpl.java

```diff
@@ -213,6 +213,11 @@ public void writeBatch(WriteBatchRequest request,
         SingleStoreConfiguration conf = new SingleStoreConfiguration(request.getConfigurationMap());
 
         try (Connection conn = JDBCUtil.createConnection(conf);) {
+            if (request.getTable().getColumnsList().stream()
+                    .allMatch(column -> !column.getPrimaryKey())) {
```

Comment:
noneMatch method is more readable I think

Reply:
Agree :)